### DISCUSSION TITLE
kick.com support

### DISCUSSION THREAD

**Discussion by NoOneNook** [Zibbp/ganymede#245] [Ideas]
Hi, have you considered adding support for kick.com?
I myself am learning how to code against the kick.com api these days

**Comment by Zibbp** [comment]
Does Kick have an official API?

Adding support for other sites YouTube, Kick, etc is a large feature and will require refactoring the backend quite a bit. I hope to eventually support it but haven't begun looking into it. The largest issue is going to be the chat for each site.

**Reply by NoOneNook** [reply]
I have a little idea how you can connect recordings to kick stream's, but I haven't figured out how to retrieve chat yet. No, as of today, kick does not have a well-functioning api system, but I and some other developers have created a separate api system that checks if kick streamers go live so we can have custom "live notifications" in discord servers.

I think you can download video for recording the stream via this link
https://kick.com/api/v1/channels/STREAMER_NAME

Demo associated with one of my streamers: https://kick.com/api/v1/channels/texazrexaz

Inside that link we find a section called "playback_url" which contains a link with an .m3u8 file which I know is the one associated with the streamer when the streamer is live. I myself have downloaded the .m3u8 file from there when a streamer was live to be able to watch the streamer's stream using Windows' program for video playback locally on my PC

I myself am waiting for kick to fix a more complete api system

**Reply by NoOneNook** [reply]
https://api.kick.com/api/v1/channels/STREAMER_NAME/chat

Demo: https://api.kick.com/api/v1/channels/texazrexaz/chat

**Reply by NoOneNook** [reply]
Could this system be a solution perhaps? haven't seen anything on it just read the readme section
https://github.com/zSoulweaver/kickerbot

''Kickerbot is designed to enhance the functionalities of Kick.com by providing developers with a powerful library (KClient) to interact with the private Kick.com API using Playwright. Additionally, it includes a chat bot (Kickerbot) built using the NestJS framework, offering a seamless integration of bot functionalities within the Kick.com ecosystem.''

**Reply by Zibbp** [reply]
The `/api/v1/channels/CHANNEL` route seems to have most of the information required. Looking at the request's it seems that there is already a `v2` endpoint as well. The problem with implementing API routes using non official sources is that they could be removed at any point. Hopefully Kick release an official API soon.

Streamlink (used to download the live and vods of Twitch) isn't supporting Kick yet due to it still being in beta and not supporting an official API.

One thing I just noticed is that Kick does not have chat playback on VODs yet which blocks implementing this as the chat is necessary for an archive in my opinion.
